app/views/layouts/application.html.erb

```diff
@@ -146,7 +146,7 @@
   <!--<![endif]-->
 
   <script>
-    <%= raw Rails.application.assets['modernizr/modernizr'] %>
+    <%= Sprockets::Railtie.build_environment(Rails.application, true)['modernizr/modernizr'].to_s %>
```

Comment:
Can we move this logic out of the view and into a helper method please.
Are there any regression tests in place to cover this?

Reply:
@scott-ad-riley I concur with @willhall88's comments about moving this piece of code into a tested helper method of some sort

Reply:
Please could you give me a bit more context as to what broke or failed when this was pushed to QA?

Reply:
Does `modernizr` need to go through sprockets? could we not just add this is a 
`javascript_tag 'modernizr/modernizr'` ?

Reply:
Please could you update the commit message to give the context as to why this change was made ( the reasons you gave above)

Reply:
@scott-ad-riley could we not do something like this, please? 

` <%= Sprockets::Railtie.build_environment(Rails.application, true)['modernizr/modernizr'].to_s %>`

```ruby
# file: app/views/layouts/application.html.erb

<%= load_modernizr %>
```

```
# file: app/helpers/application_helper.rb
...
def load_modernizr
  Sprockets::Railtie.build_environment(Rails.application, true)['modernizr/modernizr'].to_s
end
```

```
# file: spec/helper/application_helper_spec.rb

RSpec.describe ApplicationHelper, type: :helper do
  describe '#load_modernizr'
    it 'returns raw text of Modernizr library'
      ...
    end
  end
end
```

Reply:
In response to the question of 'why test this' - In my eyes if something has broken in QA, then that should have a test in place to ensure it doesn’t get that far again.

Reply:
I personally think a helper would be a better place for this line of code. Purely to try and keep logic outside of the view. I also think wrapping this into a well named helper method will give this piece of code a lot more context. 
As can be seen from this thread, it has had to be explained why this change was made and what the line of code was doing. Neither the PR description or the commit message held that info. I think it would be more helpful for a future person to understand if they see this line of code in a method which gives it more context and understanding

Reply:
Ok - adding the test and helper.